contracts/ServiceNodeContribution.sol

```diff
@@ -179,6 +191,17 @@ contract ServiceNodeContribution is Shared {
         }
     }
 
+    function contributeFundsWithPermit(
+        uint256 amount,
+        uint256 deadline,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) public {
+        IERC20Permit(address(SENT)).permit(msg.sender, address(this), amount, deadline, v, r, s);
```

Comment:
If this can be `SENT.permit(...)` I would prefer that.  If not, I'm curious as to why this (what you've put, which I assume works) works but `SENT.permit(...)` does not work.